src/Cns.ts

```diff
@@ -57,14 +54,37 @@ export default class Cns extends NamingService {
     );
   }
 
+  static async getNetworkConfigs(config?: { url: string } | { provider: Provider } ): Promise<{ network: CnsSupportedNetworks, provider: Provider } | undefined> {
+    if (!config) {
+      return undefined;
+    }
+    try {
+      const provider = hasProvider(config) ? config.provider : new FetchProvider(NamingServiceName.CNS, config.url);
+      const networkId = await provider.request({method: "net_version"}) as number;
+      const networkName = EthereumNetworksInverted[networkId];
+      if (!networkName || !CnsSupportedNetwork.guard(networkName)) {
+        throw new ConfigurationError(ConfigurationErrorCode.UnsupportedNetwork, {method: NamingServiceName.CNS});
+      }
+      return {
+        network: networkName,
+        provider
+      }
+    } catch(error) {
+      if (error instanceof FetchError) {
+        throw new ConfigurationError(ConfigurationErrorCode.IncorrectBlockchainProvider, {method: NamingServiceName.CNS});
```

Comment:
@JohnnyJumper Imagine I have temporary issues with internet connection. Should i get this error in that case? What does this error mean to me?